tools/crates-validator/src/profiles/parser/utils.rs

```diff
@@ -0,0 +1,11 @@
+use std::{fmt::Display, fs};
+
+use anyhow::{Context, Result};
+use serde_yaml::Value;
+
+pub(crate) fn read_profile(path: impl AsRef<str> + Display) -> Result<Value> {
+    let content = fs::read_to_string(path.as_ref())
+        .with_context(|| format!("Failed to read profile file: {path}",))?;
+    serde_yaml::from_str(&content)
+        .with_context(|| format!("Failed to parse yaml with content:\n{content}",))
+}
```

Comment:
Instead of deserializing the whole file as `Value`, what about just send as `String`, since we're calling `serde_yaml::from_value` two times one for batch and one for individual profile? 

Also, would be better to have a new module that combine both `batch` and `crates` key into a single struct, so we can deserialize the content a single time with a single function call.